groundDB.client.js

```diff
@@ -433,12 +434,14 @@ var _checkDocs = function _checkDocs(a) {
 
 // At some point we can do a remove all local-only data? Making sure that we
 // Only got the same data as the subscription
-var _removeLocalOnly = function _removeLocalOnly() {
+// If passing query we'll remove only those that pass it (and of course are only local)
+var _removeLocalOnly = function _removeLocalOnly(query) {
   var self = this;
+  query = query || {};
 
   _groundUtil.each(self._localOnly, function _loadDatabaseEach(isLocalOnly, id) {
     if (isLocalOnly) {
-      self._collection.remove({ _id: id });
+      self._collection.remove(_.extend({}, query, { _id: id }));
```

Comment:
Yes what you wrote is sufficient. And I prefer it over mine version.
Should I correct my pull request?